File: backend/src/main/java/com/fdb/backend/Entities/BorrowingTransaction.java
package com.fdb.backend.Entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class BorrowingTransaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate borrowingDate;
    private LocalDate returnDate;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "userID")
    private User user;

    @ManyToOne
    @JoinColumn(name = "book_isbn", referencedColumnName = "isbn")
    private Book book;
}

File: frontend/src/app/models/borrowingTransactionModel/borrowing-transaction.ts
import { Book } from "../bookModel/book";
import { User } from "../userModel/user";

export class BorrowingTransaction {
  id!: number;
  // borrowerId!: number;
  // bookIsbn!: number;
  borrowingDate!: Date;
  returnDate: Date | null = null;
  user!: User;
  book!: Book;
}
